components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java

```diff
@@ -198,8 +198,21 @@ public SAMLSSOServiceProviderDO parse(String metadata, SAMLSSOServiceProviderDO
             List<RoleDescriptor> roleDescriptors = entityDescriptor.getRoleDescriptors();
             //TODO: handle when multiple role descriptors are available
             //assuming only one SPSSO is inside the entitydescripter
-            RoleDescriptor roleDescriptor = roleDescriptors.get(0);
-            SPSSODescriptor spssoDescriptor = (SPSSODescriptor) roleDescriptor;
+            RoleDescriptor roleDescriptor;
+            SPSSODescriptor spssoDescriptor;
+
+            if (roleDescriptors.size() > 0) {
```

Comment:
shall we use CollectionUtils.isNotEmpty

Reply:
Validated using `CollectionUtils.isEmpty` in order to remove the else block.